### PR TITLE
Improve code coverage for System.IO.UnmanagedMemoryStream

### LINKED ISSUE #929: Improve code coverage for System.IO.UnmanagedMemoryStream
According to the current code coverage report, this library is only at 79%.  Our general goal is closer to 90%, though this will depend on the library.  We should investigate to see whether the current coverage level is appropriate, and if not, improve our test suites to provide better coverage.

### PR DESCRIPTION
**Pull request by Maxwe11**
Improved code coverage for `System.IO.UnmanagedMemoryStream`
Before: 79%
After: 89%
Fixes  #929

### CODE REVIEW

**Comment by joshfree** [comment]
Thanks @Maxwe11 for your contribution!  Can you share what the code coverage numbers were before and after your change?

/cc @KrzysztofCwalina

File: src/System.IO.UnmanagedMemoryStream/tests/UmsTests.cs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using Xunit;
+using System.Threading;
 
 // TODO: add WriteAsync, Timeout, Flush, CopyTo tests
```
Comment (joshfree) [inline src/System.IO.UnmanagedMemoryStream/tests/UmsTests.cs]: Would you be interested in addressing this TODO?

**Comment by Maxwe11** [comment]
Before: 79%
After: 89%

File: src/System.IO.UnmanagedMemoryStream/tests/UmsTests.cs (hunk 1 above)
Comment (Maxwe11) [inline src/System.IO.UnmanagedMemoryStream/tests/UmsTests.cs]: WriteAsync (mostly) and Flush included in this PR.
Timeout and CopyTo are not overloaded by `UnmanagedMemoryStream`, so what the reason to put their test here?

File: src/System.IO.UnmanagedMemoryStream/tests/UmaTests.cs [hunk 2]
```diff
@@ -42,93 +42,237 @@ public static void UmaInvalidReadWrite()
...
+                Assert.Equal(expectedByte, actualByte);
+                pos += sizeof(byte);
+
+                sbyte actualSByte = uma.ReadSByte(pos);
+                Assert.Equal(expectedSByte, actualSByte);
+                pos += sizeof(sbyte);
+
+                pos = EnsureAligned(pos, sizeof(char));
+                char actualChar = uma.ReadChar(pos);
+                Assert.Equal(expectedChar, actualChar);
```
Comment (stephentoub) [inline src/System.IO.UnmanagedMemoryStream/tests/UmaTests.cs]: It's not a problem, but I'm curious why you chose to define these locals, e.g. why you didn't instead save a line (and the possibility of accidentally reusing the variable inappropriately) in each of these blocks by just doing:
```
Assert.Equal(expectedChar, uma.ReadChar(pos);
```
Is there an advantage to having the local?

File: src/System.IO.UnmanagedMemoryStream/tests/UmaTests.cs [hunk 3]
```diff
@@ -42,93 +42,237 @@ public static void UmaInvalidReadWrite()
...
         }
+
+        private static long EnsureAligned(long position, int n)
+        {
+            return ((position & (n - 1)) == 0) ? position : (position / n + 1) * n;
+        }
+
+        private static long EnsureNotAligned(long position, int n)
+        {
+            return ((position & (n - 1)) != 0) ? position : ++position;
```
Comment (stephentoub) [inline src/System.IO.UnmanagedMemoryStream/tests/UmaTests.cs]: This is wrong if position is 0, right?

**Comment by stephentoub** [comment]
The additions LGTM. Thanks.

File: src/System.IO.UnmanagedMemoryStream/tests/UmaTests.cs (hunk 2 above)
Comment (Maxwe11) [inline src/System.IO.UnmanagedMemoryStream/tests/UmaTests.cs]: Reasonable, fixed.

**Comment by Maxwe11** [comment]
Failed due to #606

**Comment by stephentoub** [comment]
@dotnet-bot test this please

File: src/System.IO.UnmanagedMemoryStream/tests/UmsTests.cs (hunk 1 above)
Comment (KrzysztofCwalina) [inline src/System.IO.UnmanagedMemoryStream/tests/UmsTests.cs]: Could you add a comment about this. ... and I think you mean there are not overrides.

Comment (Maxwe11) [inline src/System.IO.UnmanagedMemoryStream/tests/UmsTests.cs]: Yes I mean "there are not overrides". Could you clarify about comment please.

File: src/System.IO.UnmanagedMemoryStream/tests/UmsTests.cs [hunk 4]
```diff
@@ -70,30 +71,34 @@ public static void LengthTests()
             {
                 UnmanagedMemoryStream stream = manager.Stream;
                 UmsTests.ReadWriteUmsInvariants(stream);
-                // TODO: uncomment once we run againts core CLR; currently type load issues with IOException
-                // Assert.Throws<IOException>(() => stream.SetLength(1001)); 
+                Assert.Throws<IOException>(() => stream.SetLength(1001)); 
                 Assert.Throws<ArgumentOutOfRangeException>(() => stream.SetLength(SByte.MinValue));
+
```
Comment (KrzysztofCwalina) [inline src/System.IO.UnmanagedMemoryStream/tests/UmsTests.cs]: Please remove the extra newlane

**Comment by KrzysztofCwalina** [comment]
Very nice. Thanks for doing this work. I just had a couple of nit comments, but I think it's good to merge.

File: src/System.IO.UnmanagedMemoryStream/tests/UmsTests.cs (hunk 4 above)
Comment (Maxwe11) [inline src/System.IO.UnmanagedMemoryStream/tests/UmsTests.cs]: removed

File: src/System.IO.UnmanagedMemoryStream/tests/UmsTests.cs (hunk 1 above)
Comment (KrzysztofCwalina) [inline src/System.IO.UnmanagedMemoryStream/tests/UmsTests.cs]: I meant: could you add a comment (at the top of this file) stating which APIs are not covered by these tests and why? Possibly it's just a simple comment about CopyTo.

**Pull request merged**